### PR TITLE
[ISSUE #2142]add distributed lock to avoid onceListener invoke multi times

### LINKED ISSUE #2142: Using AbstractDistributeOnceElasticJobListener，why every instance execute these two methods ： doBeforeJobExecutedAtLastStarted 、doAfterJobExecutedAtLastCompleted？
**version elastic-job-lite-core 3.0.1**

## Question

**For English only**, other languages will not accept.

Before asking a question, make sure you have:

- Googled your question.
- Searched open and closed [GitHub issues](https://github.com/apache/shardingsphere-elasticjob/issues).
- Read documentation: [ElasticJob Doc](https://shardingsphere.apache.org/elasticjob/current/en/overview/).

Please pay attention on issues you submitted, because we maybe need more details. 
If no response anymore and we cannot reproduce it on current information, we will **close it**.

### PR DESCRIPTION
**Pull request by xlcn**
Fixes #2142 .

Changes proposed in this pull request:

one job, multi instances completed on concurrency, can cause once listener execute some times, my current system use redis lock to avoid, but source code should add a distributed lock to avoid this, use curator InterProcessMutex now, please help review and give some advice.

### CODE REVIEW

File: elasticjob-lite/elasticjob-lite-core/src/main/java/org/apache/shardingsphere/elasticjob/lite/internal/guarantee/GuaranteeService.java [hunk 1]
```diff
@@ -17,11 +17,16 @@
 
 package org.apache.shardingsphere.elasticjob.lite.internal.guarantee;
 
+import org.apache.curator.framework.CuratorFramework;
+import org.apache.curator.framework.recipes.locks.InterProcessMutex;
```
Comment (TeslaCN) [inline elasticjob-lite/elasticjob-lite-core/src/main/java/org/apache/shardingsphere/elasticjob/lite/internal/guarantee/GuaranteeService.java]: We have decoupled registry center since 3.0.2. Using curator here directly may not acceptable.

**Review by TeslaCN — CHANGES_REQUESTED** [review]

**Comment by TeslaCN** [comment]
Is it possible to specifying an instance to execute the listener instead of using distributed lock?

**Comment by xlcn** [comment]
> 

okay

File: elasticjob-lite/elasticjob-lite-core/src/test/java/org/apache/shardingsphere/elasticjob/lite/internal/guarantee/GuaranteeServiceTest.java [hunk 2]
```diff
@@ -143,4 +152,32 @@ public void assertClearAllCompletedInfo() {
         guaranteeService.clearAllCompletedInfo();
         verify(jobNodeStorage).removeJobNodeIfExisted("guarantee/completed");
     }
+
```
Comment (TeslaCN) [inline elasticjob-lite/elasticjob-lite-core/src/test/java/org/apache/shardingsphere/elasticjob/lite/internal/guarantee/GuaranteeServiceTest.java:155]: Does it work without `@Test` annotation?

File: elasticjob-lite/elasticjob-lite-core/src/main/java/org/apache/shardingsphere/elasticjob/lite/internal/guarantee/GuaranteeService.java [hunk 3]
```diff
@@ -105,21 +109,90 @@ public boolean isRegisterCompleteSuccess(final Collection<Integer> shardingItems
...
      * @return job's sharding items are all completed or not
      */
     public boolean isAllCompleted() {
         return jobNodeStorage.isJobNodeExisted(GuaranteeNode.COMPLETED_ROOT)
-                && configService.load(false).getShardingTotalCount() <= jobNodeStorage.getJobNodeChildrenKeys(GuaranteeNode.COMPLETED_ROOT).size();
+                && configService.load(false).getShardingTotalCount()
+                <= jobNodeStorage.getJobNodeChildrenKeys(GuaranteeNode.COMPLETED_ROOT).size();
     }
-    
+
```
Comment (TeslaCN) [inline elasticjob-lite/elasticjob-lite-core/src/main/java/org/apache/shardingsphere/elasticjob/lite/internal/guarantee/GuaranteeService.java:122]: Could you revert the unrelated changes in this file?

Comment (xlcn) [reply elasticjob-lite/elasticjob-lite-core/src/main/java/org/apache/shardingsphere/elasticjob/lite/internal/guarantee/GuaranteeService.java:122]: Whitespace can not revert, no diff, git can not push, could be problem of line break format, my format is LF. Do I need to handle it in other ways?

File: elasticjob-lite/elasticjob-lite-core/src/test/java/org/apache/shardingsphere/elasticjob/lite/internal/guarantee/GuaranteeServiceTest.java (hunk 2 above)
Comment (xlcn) [reply elasticjob-lite/elasticjob-lite-core/src/test/java/org/apache/shardingsphere/elasticjob/lite/internal/guarantee/GuaranteeServiceTest.java:155]: > Does it work without `@Test` annotation?

done

File: elasticjob-lite/elasticjob-lite-core/src/main/java/org/apache/shardingsphere/elasticjob/lite/internal/guarantee/GuaranteeService.java (hunk 3 above)
Comment (TeslaCN) [reply elasticjob-lite/elasticjob-lite-core/src/main/java/org/apache/shardingsphere/elasticjob/lite/internal/guarantee/GuaranteeService.java:122]: Got it. Let me handle it later.

**Review by TeslaCN — APPROVED** [review]
